perfkitbenchmarker/pkb.py

```diff
@@ -257,8 +255,8 @@ def RunBenchmark(benchmark, collector, sequence_number, total_benchmarks):
     try:
       with end_to_end_timer.Measure('End to End'):
         if FLAGS.run_stage in [STAGE_ALL, STAGE_PREPARE]:
-          spec = DoPreparePhase(
-              benchmark, benchmark_info, benchmark_name, detailed_timer)
+          spec = benchmark_spec.BenchmarkSpec(benchmark_info)
```

Comment:
Please add a comment on why this separation is important.

By getting the spec you have also caused it to be saved for future cleanup.